commons/zenoh-protocol/src/core/endpoint.rs

```diff
@@ -385,8 +385,9 @@ impl ConfigMut<'_> {
             self.0.address(),
             self.0.metadata(),
             parameters::from_iter(parameters::sort(parameters::join(
-                self.0.config().iter(),
+                // Endpoint config overwrites incoming config
                 iter.map(|(k, v)| (k.borrow(), v.borrow())),
+                self.0.config().iter(),
```

Comment:
I don't think this is the right change to solve the issue. 
IMHO this changes is wrong w.r.t. the expected behaviour of this function where the iter in this is overwritten by the config. 

The change should be done in the transport like here: https://github.com/eclipse-zenoh/zenoh/blob/233240622f548c18fe41450144010ad1b7e1fe17/io/zenoh-transport/src/unicast/manager.rs#L386.

Reply:
This function is currently only used for merging config file with endpoint config. Would renaming it to indicate its purpose solve the issue then?